### PR TITLE
Automate changelog and manifest.xml creation

### PR DESCRIPTION
**Pull request by Wires77**
We'll probably want to adjust the wording in the PRs, and we may have to play around with the parameters until we find something we're happy with, but I thought I'd throw this up for discussion.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+            - name: Install changelog generator
+              run: sudo gem install github_changelog_generator
+            - name: Generate Release notes
+              run: github_changelog_generator --user ${{ github.repository_owner }} --project ${{ github.event.repository.name }} -t ${{ secrets.GITHUB_TOKEN }} -b CHANGELOG.md --release-branch dev --exclude-labels invalid,duplicate --future-release ${{ github.event.inputs.releaseVersion }} --since-tag ${{ github.event.inputs.sinceTag }} --no-issues true
+            - name: Update manifest.xml
+              run: python3 update_manifest.py; cp manifest-updated.xml manifest.xml
```
Comment (ppoelzl) [inline .github/workflows/release.yml:24]:
```suggestion
              run: python3 update_manifest.py --quiet --in-place --set-version=${{ github.event.inputs.releaseVersion }}
```
Updated script invocation for new arguments.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Release next version
+on: 
+  workflow_dispatch:
+      inputs:
+          sinceTag:
+              description: 'Gather updates from this tag until now'
+              required: true
+              default: 'v2.0.x'
```
Comment (ppoelzl) [inline .github/workflows/release.yml:8]: "v2.0.x" is not SemVer. See https://semver.org/#is-v123-a-semantic-version

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+              uses: peter-evans/create-pull-request@v3.8.0
+              with:
+                  draft: true
+                  title: Release ${{ github.event.inputs.releaseVersion }}
+                  branch: release_${{ github.event.inputs.releaseVersion }}
+                  body: |
+                      Draft release of ${{ github.event.inputs.releaseVersion }}
+
+                      Edits will be made as necessary to prepare the codebase for release
+                  commit-message: 'Changelogs and manifest.xml updated for ${{ github.event.inputs.releaseVersion }}'
```
Comment (ppoelzl) [inline .github/workflows/release.yml:35]:
```suggestion
                  commit-message: 'Prepare release ${{ github.event.inputs.releaseVersion }}'
```
Shorter commit messages are preferable.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            - name: Create Pull Request
+              uses: peter-evans/create-pull-request@v3.8.0
+              with:
+                  draft: true
+                  title: Release ${{ github.event.inputs.releaseVersion }}
+                  branch: release_${{ github.event.inputs.releaseVersion }}
+                  body: |
+                      Draft release of ${{ github.event.inputs.releaseVersion }}
+
+                      Edits will be made as necessary to prepare the codebase for release
```
Comment (ppoelzl) [inline .github/workflows/release.yml:34]:
```suggestion
                      Draft release of ${{ github.event.inputs.releaseVersion }}.

                      Edits will be made as necessary to prepare the codebase for release.
```
Added punctuation.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            - name: Generate Release notes
+              run: github_changelog_generator --user ${{ github.repository_owner }} --project ${{ github.event.repository.name }} -t ${{ secrets.GITHUB_TOKEN }} -b CHANGELOG.md --release-branch dev --exclude-labels invalid,duplicate --future-release ${{ github.event.inputs.releaseVersion }} --since-tag ${{ github.event.inputs.sinceTag }} --no-issues true
+            - name: Update manifest.xml
+              run: python3 update_manifest.py; cp manifest-updated.xml manifest.xml
+            - name: Create Pull Request
+              uses: peter-evans/create-pull-request@v3.8.0
+              with:
+                  draft: true
+                  title: Release ${{ github.event.inputs.releaseVersion }}
+                  branch: release_${{ github.event.inputs.releaseVersion }}
```
Comment (ppoelzl) [inline .github/workflows/release.yml:30]:
```suggestion
                  branch: release-${{ github.event.inputs.releaseVersion }}
```
GitHub branches should be kebab-case.

**Review by ppoelzl — CHANGES_REQUESTED** [review]

**Comment by Wires77** [comment]
In testing my personal branch, I'm unsure if this workflow is able to be invoked manually until it's in the main branch, unfortunately, which makes it pretty hard to test.  That's really the only trigger that makes sense, too, since we only want to do this once, right before the merge to master.  We can potentially run these same steps locally before the next release, and make the PR, to ensure we have the settings right before merging it into master

**Review by ppoelzl — APPROVED** [review]
LGTM, I don't need we need to verify this manually, as long as the PR it produces is ok.

**Pull request merged**